### PR TITLE
duplicate arrays using .drop(0) instead of .dup

### PR DESCRIPTION
**Pull request by mojavelinux**
Using .drop(0) is 1.5x as fast as .dup.

### CODE REVIEW

**Comment by Mogztter** [comment]
The performance on Asciidoctor.js should be identical since Opal was already using the same strategy with `slice(0)`: https://github.com/opal/opal/blob/b575f088adac3c37a577c7f7b290ad3a5dddb5d7/opal/corelib/array.rb#L872

File: CHANGELOG.adoc [hunk 1]
```diff
@@ -38,6 +38,7 @@ Improvements::
   * allow hyphen to be used custom block macro name as long as it's not the first character (#2620)
   * use shorthands %F and %T instead of %Y-%m-%d and %H:%M:%S to format time
   * read file in binary mode whenever contents are being normalized
+  * use .drop(0) to duplicate arrays (rougly 1.5x as fast as .dup)
```
[the remark this answers is not in the document]
Comment (mojavelinux) [reply CHANGELOG.adoc]: doh! yep

**Comment by mojavelinux** [comment]
Nice! I really can't explain why this is so much faster in CRuby, but the difference is very clear.

**Comment by mojavelinux** [comment]
And actually, I found that slice(0) is the fastest way to dup an array in JavaScript. Interesting how it's parallel.

**Pull request merged**